elide-async/src/main/java/com/yahoo/elide/async/service/DefaultAsyncQueryDAO.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.service;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.UUID;
+
+import javax.inject.Singleton;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+
+import com.yahoo.elide.Elide;
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.AsyncQueryResult;
+import com.yahoo.elide.core.DataStore;
+import com.yahoo.elide.core.DataStoreTransaction;
+import com.yahoo.elide.core.RequestScope;
+import com.yahoo.elide.jsonapi.models.JsonApiDocument;
+import com.yahoo.elide.request.EntityProjection;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Utility class which implements AsyncQueryDAO
+ */
+@Singleton
+@Slf4j
+public class DefaultAsyncQueryDAO implements AsyncQueryDAO {
+
+    private Elide elide;
+    private DataStore dataStore;
+
+    public DefaultAsyncQueryDAO() {}
+
+    public DefaultAsyncQueryDAO(Elide elide, DataStore dataStore) {
+    	this.elide = elide;
+    	this.dataStore = dataStore;
+    }
+
+    @Override
+    public void setElide(Elide elide) {
+        this.elide = elide;
+    }
+
+    @Override
+    public void setDataStore(DataStore dataStore) {
+        this.dataStore = dataStore;
+    }
+
+    @Override
+    public AsyncQuery updateAsyncQuery(UUID asyncQueryId, UpdateQuery updateFunction) {
+        log.debug("updateAsyncQuery");
+        AsyncQuery queryObj = (AsyncQuery) executeInTransaction(dataStore, (tx, scope) -> {
+            EntityProjection asyncQueryCollection = EntityProjection.builder()
+                    .type(AsyncQuery.class)
+                    .build();
+            AsyncQuery query = (AsyncQuery) tx.loadObject(asyncQueryCollection, asyncQueryId, scope);
+            updateFunction.update(query);
+            tx.save(query, scope);
+            return query;
+        });
+        return queryObj;
+    }
+
+    @Override
+    public Iterable<Object> updateAsyncQueryCollection(Iterable<Object> asyncQueryList, UpdateQuery updateFunction) {
+        log.debug("updateAsyncQueryCollection");
+        executeInTransaction(dataStore, (tx, scope) -> {
+            EntityProjection asyncQueryCollection = EntityProjection.builder()
+                    .type(AsyncQuery.class)
+                    .build();
+
+            Iterator<Object> itr = asyncQueryList.iterator();
+            while(itr.hasNext()) {
+                AsyncQuery query = (AsyncQuery) itr.next();
+                AsyncQuery asyncQuery = (AsyncQuery) tx.loadObject(asyncQueryCollection, query.getId(), scope);
```

Comment:
I don't think this function should load anything.  Load can be a separate function.

Reply:
Same comment still applies.  All of these objects have already been loaded.  This will result in a lot of unnecessary calls to the DB.  

Instead of loading them, just call the updateFunction and save.